### ISSUE TITLE
www.myabandonware.com uBlock filters breakage/issue

### ISSUE THREAD

**Issue by Silent-Soldier** [uBlockOrigin/uBlock-issues#2292]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

`https://www.myabandonware.com/`

### Describe the issue

myabandonware.com fails to render text with default/OOB settings.

### uBlock Origin version

Latest Dev 1.44.5b4

### Browser name and version

Firefox Nightly 107.0a1

### Settings

Disabled ALL default/built-in and third party filter lists.
Removed ALL custom filters and rules.
Basically made uBlockOrigin a blank slate, before testing.

### Notes

Relates to [Commit 4e86ea4](https://github.com/uBlockOrigin/uAssets/commit/4e86ea4106148f824a0b9c111df3c6fc8f049f9b). 

Disabling all filters (built-in and 3rd party) allows website to render. 

Through process of elimination, adding the individual myabandonware.com filters from uBlock filters to My Filters pane, and disabling each one-by-one, leads to ```myabandonware.com##div:matches-css-after(content:/Ads/)``` being the culprit. Commented out/deleted, the website renders normally, otherwise no text with background rendering only.

**Comment by mapx-** [comment]
@gorhill

**Comment by peace2000** [comment]
Now with the new csstree library I experience randomly websites going blank, and it seems to be related to cosmetic filtering as when I disable it, the sites start showing. Can't identify a specific filter where this is related, when I disable / enable lists the sites might start working correctly again and they stay that way even if I re-enable the list(s) I previously disabled. Just saying this in case someone else experiences something similar.

This happened e.g. with https://www.mikrobitti.fi/ and https://outlook.live.com/ (I was logged in here)

Wouldn't otherwise post this but I had this issue on my main operaring system and with my virtual machine. Sorry I can't give clear repro steps for this.

**Comment by mapx-** [comment]
@peace2000 I cannot reproduce blank page issue on those 2 pages

**Comment by peace2000** [comment]
@mapx- That's good though others are saying the same here https://www.reddit.com/r/uBlockOrigin/comments/xog9s9/ublock_origin_1445b3_on_redditcom_causes_white/

**Comment by mapx-** [comment]
@peace2000 if you still can reproduce the issue provide eventually the specific page where that happens

**Comment by peace2000** [comment]
I opened my old Firefox nightly installation on my virtual machine and directly updated to the latest uBO (not emptying any cache etc.) and `https://www.mikrobitti.fi/` doesn't work properly on that browser. It's not entirely blank but it only loads top area:

![kuva](https://user-images.githubusercontent.com/17256841/192331348-46a78621-2318-4b67-8017-d18f7b542ebb.png)

**Comment by gorhill** [comment]
Use the DOM inspector to find out which filter is causing this.

**Comment by peace2000** [comment]
> Use the DOM inspector to find out which filter is causing this.

`www.mikrobitti.fi##div[class]:has(> [width] > [id|="almad"]:not([id|="almad-mobileParade"]))`

**Comment by gorhill** [comment]
I tried `www.mikrobitti.fi##div[class]:has(> [width] > [id|="almad"]:not([id|="almad-mobileParade"]))` in various versions of uBO (1.44.4, 1.44.5b5, 1.44.5b0) and I cannot reproduce.

**Comment by gorhill** [comment]
It was probably an issue with 1.44.2.

**Comment by gorhill** [comment]
Yes, filters are compiled and the compiled representation is used, so you were probably using the old compiled representation of `:not()` in the newer version of uBO -- until the list was updated, in which case all filters are recompiled.

**Comment by peace2000** [comment]
Would it be possible to force the filter cache to be renewed when uBO updates to newer version?

**Comment by gorhill** [comment]
To be accurate, the issue is not really related to cssTree, it's me who removed code that was still needed, this would have occurred cssTree or not.

**Comment by gorhill** [comment]
> Would it be possible to force the filter cache to be renewed when uBO updates to newer version?

Yes, there is way, but usually I tried to avoid this if possible. In the current case anyway, the advice for site breakage is too force an update of filter lists, which would also solve the issue.

**Comment by u-RraaLL** [comment]
> Now with the new csstree library I experience randomly websites going blank, and it seems to be related to cosmetic filtering as when I disable it, the sites start showing. Can't identify a specific filter where this is related to, when I disable / enable lists the sites might start working correctly again and they stay that way even if I re-enable the list(s) I previously disabled. Just saying this in case someone else experiences something similar.

I had this issue twice or so on various websites. Yesterday when I just installed 1.44.5b3. And today on 1.44.5b5. I've also dropped down to b0 and back up to b3 and 5 multiple times to test another issue I'm having and I've only encountered the problem twice.

Opening DOM Inspector didn't bring the elements back though. Only disabling CF and refreshing showed whole websites again. I think I had to purge and update filters multiple times to fix the issue. EL didn't seem truncated though.

Anyway, I see there was a commit that didn't make it to a new release yet, so I've no idea whether the issue will return.
